components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -2954,6 +2954,82 @@ export function createWalletApi () {
         providesTags: (res, er, arg) => [
           { type: 'SolanaEstimatedFees', id: arg.txId }
         ]
+      }),
+      getEthTokenDecimals: query<
+        number,
+        Pick<BraveWallet.BlockchainToken, 'chainId' | 'contractAddress'>
+      >({
+        queryFn: async (arg, { endpoint }, extraOptions, baseQuery) => {
+          try {
+            const { jsonRpcService } = baseQuery(undefined).data
+            const { errorMessage, decimals } =
+              await jsonRpcService.getEthTokenDecimals(
+                arg.contractAddress,
+                arg.chainId
+              )
+
+            if (!decimals) {
+              throw new Error(errorMessage)
+            }
+
+            return {
+              data: Number(decimals)
+            }
+          } catch (error) {
+            return handleEndpointError(
+              endpoint,
+              `Unable to fetch token decimals for ${arg.contractAddress}`,
+              error
+            )
+          }
+        },
+        providesTags: (res, err, arg) =>
+          err
+            ? ['UNKNOWN_ERROR']
+            : [
+                {
+                  type: 'EthTokenDecimals',
+                  id: [arg.chainId, arg.contractAddress].join('-')
+                }
+              ]
+      }),
+      getEthTokenSymbol: query<
+        string,
+        Pick<BraveWallet.BlockchainToken, 'chainId' | 'contractAddress'>
+      >({
+        queryFn: async (arg, { endpoint }, extraOptions, baseQuery) => {
+          try {
+            const { jsonRpcService } = baseQuery(undefined).data
+            const { errorMessage, symbol } =
+              await jsonRpcService.getEthTokenSymbol(
+                arg.contractAddress,
+                arg.chainId
+              )
+
+            if (!symbol) {
+              throw new Error(errorMessage)
+            }
```

Comment:
shouldnt we just check if there is an error message?

Reply:
https://github.com/brave/brave-core/pull/20203/commits/e075c8e2829179099b2af490db06926db1b88000